browser/src/Services/Explorer/ExplorerView.tsx

```diff
@@ -269,11 +265,72 @@ export interface IExplorerViewProps extends IExplorerViewContainerProps {
     updated: string[]
 }
 
-import { SidebarEmptyPaneView } from "./../../UI/components/SidebarEmptyPaneView"
+interface ISneakableNode extends IExplorerViewProps {
+    node: Node
+    selectedId: string
+}
 
-import { commandManager } from "./../CommandManager"
+const SneakableNode = ({ node, selectedId, ...props }: ISneakableNode) => {
+    const handleClick = () => {
+        props.onClick(node.id)
+    }
+
+    return (
+        <Sneakable callback={handleClick}>
+            <NodeView
+                node={node}
+                isSelected={node.id === selectedId}
+                isCreating={props.isCreating}
+                onCancelCreate={props.onCancelCreate}
+                onCompleteCreate={props.onCompleteCreate}
+                onCompleteRename={props.onCompleteRename}
+                isRenaming={props.isRenaming}
+                onCancelRename={props.onCancelRename}
+                updated={props.updated}
+                yanked={props.yanked}
+                moveFileOrFolder={props.moveFileOrFolder}
+                onClick={handleClick}
+            />
+        </Sneakable>
+    )
+}
+
+const ExplorerContainer = styled.div`
+    height: 100%;
+    ${enableMouse};
+`
 
 export class ExplorerView extends React.PureComponent<IExplorerViewProps, {}> {
+    private _virtualizedList: List
+
+    public componentDidUpdate(prevProps: IExplorerViewProps) {
+        if (
+            prevProps.isCreating !== this.props.isCreating ||
+            prevProps.isRenaming !== this.props.isRenaming
+        ) {
+            this._virtualizedList.recomputeRowHeights()
+            this._virtualizedList.forceUpdate()
+        }
+    }
+
+    public calculateHeight = (selectedId: string) => ({ index }: { index: number }) => {
+        const node = this.props.nodes[index]
+        const isSelected = node.id === selectedId
+
+        const renameInProgress =
+            this.props.isRenaming.name === node.name && !this.props.isCreating && isSelected
+        const creationInProgress = this.props.isCreating && isSelected && !renameInProgress
+
+        switch (true) {
+            case renameInProgress:
+                return 35
```

Comment:
Hmm, how do we determine these sizes? I'm wondering if they are dependent on the `ui.fontSize` setting? I'm just worried about a bug where different font sizes cause rendering quirks.

Reply:
I actually fiddled with the sizes here and tbh the css with `virtualized` involved is more or less opaque to me but I didn't notice a big difference using the values which were trial and error as a result of that process, I've tweaked it however to use `ui.fontSize` to derive the size in case